plugin-server/src/worker/ingestion/hooks.ts

```diff
@@ -28,31 +28,36 @@ export function determineWebhookType(url: string): WebhookType {
     return WebhookType.Teams
 }
 
-export function getUserDetails(event: PostIngestionEvent, siteUrl: string, webhookType: WebhookType): [string, string] {
-    const userName = stringify(
-        event.person_properties?.email ||
-            event.person_properties?.name ||
-            event.person_properties?.username ||
-            event.distinctId
-    )
-    let userMarkdown: string
+export function toWebhookLink(text: string | null, url: string, webhookType: WebhookType): [string, string] {
+    const name = stringify(text)
+    let markdown: string
     if (webhookType === WebhookType.Slack) {
-        userMarkdown = `<${siteUrl}/person/${event.distinctId}|${userName}>`
+        markdown = `<${url}|${name}>`
     } else {
-        userMarkdown = `[${userName}](${siteUrl}/person/${event.distinctId})`
+        markdown = `[${name}](${url})`
```

Comment:
This seems vulnerable to crafted events like `$pageview](https://malicious-url?`, which would then link to an arbitrary url in the Discord message.

Reply:
Hmm, true. This probably was an issue already before, but I added some escaping code now. Didn't test it yet 😅